6MI0800193/Week1/Task2.cpp

```diff
@@ -0,0 +1,98 @@
+#include <iostream>
+using namespace std;
+
+struct Tuple {
+    int first;
+    int second;
+};
+
+void read_tuple(Tuple& x)
+{
+    cin >> x.first >> x.second;
+}
+
+bool compare_tuples(const Tuple& x, const Tuple& y) // x > y -> true
+{
+    return (x.first > y.first) || ((x.first == y.first) && (x.second > y.second));
+}
+
+Tuple get_biggest(const Tuple tuples[100])
+{
+    Tuple biggest = tuples[0];
+    for (size_t i = 1; i < 100; i++)
+    {
+        if (compare_tuples(tuples[i], biggest))
+        {
+            biggest = tuples[i];
+        }
+    }
+    return biggest;
+}
+
+void swap(Tuple& x, Tuple& y)
+{
+    Tuple temp = x;
+    x = y;
+    y = temp;
+}
+
+void sort_tuples(Tuple* tuples, size_t elements)
+{
+    int min_idx = 0;
+    for (size_t i = 0; i < elements - 1; i++)
+    {
+        min_idx = i;
+        for (size_t j = i + 1; j < elements; j++)
+        {
+            if (compare_tuples(tuples[min_idx], tuples[j]))
+                min_idx = j;
+        }
+
+        if (min_idx != i)

```

Comment:
What you're doing here is nice but it's also what people tend to call `micro-optimization`. If it was me doing the implementation - I'd skip the if and invoke `swap` directly.